src/cli/log.rs

```diff
@@ -129,7 +129,7 @@ impl NotificationLevel {
 /// A [`tracing::Subscriber`] [`Layer`][`tracing_subscriber::Layer`] that corresponds to Rustup's
 /// optional `opentelemetry` (a.k.a. `otel`) feature.
 #[cfg(feature = "otel")]
-fn telemetry<S>(process: &Process) -> impl Layer<S>
+fn telemetry<S>(process: &Process) -> impl Layer<S> + use<S>
```

Comment:
@djc So basically `impl Layer<S> + use<S>` means "this `impl Layer<S>` has nothing to do with `'process`", and the same thing goes for other fixes in this commit.